go/base/tasks.py

```diff
@@ -26,6 +26,9 @@ def get_and_reset_recent_conversations(vumi_api):
     redis = vumi_api.redis.sub_manager(
         ConversationMetricsMiddleware.SUBMANAGER_PREFIX)
 
+    # Reset the local cache for recent conversations
+    ConversationMetricsMiddleware.LOCAL_RECENT_CONVS = set()
```

Comment:
Since we can't clear the set here, we'll need to clear it somewhere else. I suggest that each middleware clear its own set. Twisted has a nice helper for running a function periodically -- https://twistedmatrix.com/documents/16.0.0/api/twisted.internet.task.LoopingCall.html.